test/e2e/service.go

```diff
@@ -112,25 +116,25 @@ var _ = Describe("Services", func() {
 		addEndpointPodOrFail(c, ns, name1, labels, []api.ContainerPort{{ContainerPort: 80}})
 		names = append(names, name1)
 
-		validateEndpointsOrFail(c, ns, serviceName, map[string][]int{name1: {80}})
+		validateEndpointsOrFail(c, ns, serviceName, map[string][]string{name1: {portname1}})
```

Comment:
I'm not quite sure I understand - is this because the port numbers in the pod.spec.containers[].ports[].port are not th actual port numbers exposed by endpoints (because of your host port remapping?)

This is a pretty significant change to this test and likely to get broken - most ports are NOT named, especially in tests - can you comment this requirement near the place we create pods in this test?